parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -109,11 +109,24 @@ if [[ "${GPU_NODE}" = true ]]; then
     fi
     ensureGPUDrivers
     if [[ "${ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED}" = true ]]; then
+        if [[ "${MIG_NODE}" == "true" ]] && [[ -f "/etc/systemd/system/nvidia-device-plugin.service" ]]; then
+            wait_for_file 3600 1 /etc/systemd/system/nvidia-device-plugin.service.d/10-mig_strategy.conf || exit $ERR_FILE_WATCH_TIMEOUT
+        fi
         systemctlEnableAndStart nvidia-device-plugin || exit $ERR_GPU_DEVICE_PLUGIN_START_FAIL
     else
         systemctlDisableAndStop nvidia-device-plugin
     fi
 fi
+# If it is a MIG Node, enable mig-partition systemd service to create MIG instances
+if [[ "${MIG_NODE}" == "true" ]]; then
+    REBOOTREQUIRED=true
+    ensureMigPartition
+else
+    if dpkg -s "cuda-drivers-fabricmanager-$GPU_FABRICMGR_VERSION"; then
```

Comment:
Are there restrictions on which SKUs support MIG nodes (I assume it's only the GPU skus)? If so, which section of the code would those constraints be present in?

Reply:
yes, RP handles that, but the GPU optimized vhd will have these components for non-MIG which isn't so great. I'd rather remove than add at runtime though, it'll be a bit dirty either way sadly :/